### PR TITLE
CCIP fee fix

### PR DESCRIPTION
**Pull request by gusjavaz**
Additionally this PR:

~~Replaces XERC20 extension with GXT (GlacisCrossChainToken)~~
~~Adds counterpart support to GXT~~

1. Swaps constructor arguments on LZ Adapter to follow the rest of the adapters approach
2. Force retriable to true when sendind tokens
3. Replaces setAdapterChain on CCIP adapter to  setGlacisChainIds to follow the rest of the adapters approach
4. Updates node to 18 on github action
5. Adds a Dockerfile so tests can work in a common container architecture

### CODE REVIEW

File: contracts/mediators/GlacisTokenMediator.sol
Comment (jboetticher) [inline contracts/mediators/GlacisTokenMediator.sol]: If we are keeping this as true always, then the interface should be reflected as such and this parameter should be removed

**Review by jboetticher — CHANGES_REQUESTED** [review]
Don't have much of an issue with enforcing the remote counterpart interface within the Glacis XERC20. Main issue is with the retry functionality change.

I think we should really approach Jacob & any other potential users of XERC20 with the question about enforcing the counterpart manager. Especially the tradeoff:

1. Counterpart is stored within the GXT: XERC20s must have either the GXT interface or the same address value, otherwise Glacis cannot bridge your token.
2. Counterpart is stored within the token router: all XERC20s with just the base token can be transferred, but addresses are centralized.

We can always replace/upgrade the token router later to include functionality 2 once we have correct governance protocols in place.

File: contracts/mediators/GlacisTokenMediator.sol
Comment (gusjavaz) [reply contracts/mediators/GlacisTokenMediator.sol]: You are right. Changed.

**Comment by gusjavaz** [comment]
> w
There is a third option which is to have a cross chain token registry like LZ. After reviewing all the options, my opinion is that Glacis should implement an xERC20 address registry, since we should not let Glacis rely completely on the issuer, since the issuer could change the token variant at any time. I guess for now we should only rely on a permissioned Glacis Cross Chain Token Registry where glacis labs can attest the token counterparts. I have modified the PR to reflect this proposal. What do you think?

**Comment by gusjavaz** [comment]
I've also added a Dockerfile so we can both work in a common container.

File: contracts/token/GlacisCrossChainTokenRegistry.sol
Comment (jboetticher) [inline contracts/token/GlacisCrossChainTokenRegistry.sol]: How do people permissionlessly deploy cross-chain tokens if this functionality is limited to just the owner?

**Review by jboetticher — CHANGES_REQUESTED** [review]
I can't approve this architecture if we haven't addressed the governance issue. XERC20 is supposed to give the developer more freedom and control, but if we gatekeep who can set which tokens correspond to which addresses and/or give ourselves an easy backdoor then it defeats the purpose.

Perhaps there is an architecture that sets the registry as the "last resort" instead of a requirement?

**Comment by jboetticher** [comment]
> cross chain token registry like LZ

I didn't know LZ had that, I thought that each OFT had to register their own token variants. But I'm certain that Wormhole had this originally with their token attestation. 

> the issuer could change the token variant at any time. 

If they want to do that, then they should have the ability to do so. Perhaps they are migrating to a new version of their token.

File: contracts/token/GlacisCrossChainTokenRegistry.sol
Comment (gusjavaz) [reply contracts/token/GlacisCrossChainTokenRegistry.sol]: The token deployment is permissionless but the register in Glacis is done by Glacis Governance

**Comment by gusjavaz** [comment]
> > cross chain token registry like LZ
> 
> I didn't know LZ had that, I thought that each OFT had to register their own token variants. But I'm certain that Wormhole had this originally with their token attestation.
> 
> > the issuer could change the token variant at any time.
> 
> If they want to do that, then they should have the ability to do so. Perhaps they are migrating to a new version of their token.

The Token attestation service 

> > cross chain token registry like LZ
> 
> I didn't know LZ had that, I thought that each OFT had to register their own token variants. But I'm certain that Wormhole had this originally with their token attestation.
> 
> > the issuer could change the token variant at any time.
> 
> If they want to do that, then they should have the ability to do so. Perhaps they are migrating to a new version of their token.

How Glacis could deal with a malicious issuer that changes the token counterpart after setting limits for Glacis?

File: contracts/token/GlacisCrossChainTokenRegistry.sol
Comment (jboetticher) [reply contracts/token/GlacisCrossChainTokenRegistry.sol]: Deployment is permissionless but we control when & if you can use your deployment? Sounds like the same thing to me :)

**Comment by jboetticher** [comment]
> How Glacis could deal with a malicious issuer that changes the token counterpart after setting limits for Glacis?

I'm not sure what the attack is that you're talking about here. Could you expand?

**Comment by gusjavaz** [comment]
> > How Glacis could deal with a malicious issuer that changes the token counterpart after setting limits for Glacis?
> 
> I'm not sure what the attack is that you're talking about here. Could you expand?

I'll create a separate PR for the token counterpart since it is not related to these fixes. There we can discuss the best approach.

**Comment by gusjavaz** [comment]
Removed from this PR

1. Replaces XERC20 extension with GXT (GlacisCrossChainToken)
2. Adds counterpart support to GXT

File: test/LocalTestSetup.sol
Comment (jboetticher) [inline test/LocalTestSetup.sol:151]: Will have to make change to deployment too

**Review by jboetticher — APPROVED** [review]

**Pull request merged**
